### ISSUE TITLE
[$125] Chat - With quote markdown, url text on tapping enter key not moved to next line.

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#54522]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.78-1
**Reproducible in staging?:** Yes
**Reproducible in production?:** Yes
**Issue reported by:** Applause Internal Team
**Device used:** Redmi note 10s Android 13
**App Component:** Chat Report View

## Action Performed:
1. Go to https://staging.new.expensify.com/home
2. Open chat
3. Enter > jsjjd.com
4. Send the message
5. Edit the message
6. Tap enter key and try to make url text into two lines

## Expected Result:
With quote markdown, url text on tapping enter key must be moved to next line.

## Actual Result:
With quote markdown, url text on tapping enter key not moved to next line.

## Workaround:
Unknown

## Platforms:
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/2b9c7943-2f10-48ce-b7bd-bc4efd4da2f0

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~021871675231934377672</li>
        <li>Upwork Job ID: 1871675231934377672</li>
        <li>Last Price Increase: 2024-12-31</li>
<li>Automatic offers: </li>
<ul>
<li>daledah | Contributor | 105551577</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @sakluger</details>

**Comment by sakluger** [comment]
I tested myself in the iOS app and I saw different behavior from the video above. I tested with `hshs.com` and `google.com`.

With `hshs.com`, it kept everything on the same line:

![hshs link](https://github.com/user-attachments/assets/bc5debab-e864-4b40-bb85-bf0692dadd13)

With `google.com`, which was too big to have on a single line, it kept the `https://` on the first line and moved the remainder to the second line:

![google link](https://github.com/user-attachments/assets/6b1f0eed-179b-4c9c-b0d7-d144aa8f14fc)

I also tested on MacOS chrome and could not reproduce.

**Comment by sakluger** [comment]
I'm going to set the bounty to $125 to start, hopefully this is a simple fix.

**Comment by tomekzaw** [comment]
Responded here: https://github.com/Expensify/react-native-live-markdown/pull/439#discussion_r1897303680

**Comment by sakluger** [comment]
I think @getusha is out on vacation. Let's hold until the new year.

**Comment by sakluger** [comment]
@getusha welcome back! Let me know if you need to have some of your assigned issues redistributed, I'm happy to assign this to someone else. Otherwise, I'll wait for your review.

**Comment by getusha** [comment]
@sakluger will review in short. happy new year!

**Comment by getusha** [comment]
@daledah's [proposal](https://github.com/Expensify/App/issues/54522#issuecomment-2561597708) looks good to me
🎀 👀 🎀 C+ Reviewed!

**Comment by thienlnam** [comment]
Let's go with the solution that @tomekzaw recommended - so the optional in the proposal

**Comment by tomekzaw** [comment]
PR https://github.com/Expensify/react-native-live-markdown/pull/593 has been merged into react-native-live-markdown.

**Comment by thienlnam** [comment]
Nope, looks like this has passed the 7 day prod regression period.

cc @sakluger Could you handle payment for this issue?

**Comment by sakluger** [comment]
Oops, sorry - I missed this because it was Monthly. Handling now.

**Comment by sakluger** [comment]
@daledah thanks! I paid out your Upwork contract.

I'm going to assume that there are no other outstanding payments here. @getusha - please post in Slack if you do require any payment for this issue.
